### PR TITLE
extra-custom-distro snippets  support #1859

### PR DESCRIPTION
**Pull request by DiegoAccorinti**
The idea is this: If a gnu / linux distribution that includes GEANY wants to include some customized snippets, just add the "snippets-distro.conf" file next to "snippets.info" of the system with a package. snippets will be added, to then load the user's personal in /home/user/.config/geany
It is a way for those who distribute Geany to incorporate their own snippets. In the case of Huayra gnu / linux, we added a pair that are used with the PILAS-ENGINE framework, for game development.

### CODE REVIEW

File: src/editor.c [hunk 1]
```diff
@@ -149,6 +149,39 @@ static void snippets_load(GKeyFile *sysconfig, GKeyFile *userconfig)
 	}
 	g_strfreev(groups_sys);
 
+	/* now read defined completions in DISTRO's configuration directory and add / replace them */
```
Comment (johndescs) [inline src/editor.c]: This duplicates a lot of code.

File: src/editor.c [hunk 2]
```diff
@@ -4748,10 +4791,6 @@ on_editor_scroll_event(GtkWidget *widget, GdkEventScroll *event, gpointer user_d
 {
 	GeanyEditor *editor = user_data;
 
-	/* we only handle up and down, leave the rest to Scintilla */
```
Comment (johndescs) [inline src/editor.c]: What's the point of this? Maybe belongs to some other modification set?

**Comment by elextr** [comment]
In general if a distro has extra snippets it thinks are generally useful it would be better if it made them to the system `snippets.conf` _and contributed them upstream_ rather than keeping them to themselves.

If a distro has additions that are specific to its particular focus (such as this case would appear) as @johndescs commented, it can modify `snippets.conf` as part of its building Geany packages.  Most distros make and track modifications to suit their particular requirements.

**Comment by DiegoAccorinti** [comment]
ok, tnks

**Pull request closed without merging**